test/ffmpeg-d3d11/decode/10bit/hevc.py

```diff
@@ -16,6 +16,9 @@ def before(self):
     # default metric
     self.metric = dict(type = "ssim", miny = 1.0, minu = 1.0, minv = 1.0)
     self.caps   = platform.get_caps("decode", "hevc_10")
+    # ffmpeg-d3d11 only support P010 cases
+    self.caps['fmts'].clear()
+    self.caps['fmts'].append('P010')
```

Comment:
Hmm... I think you may need to do a copy.deepcopy on return value from platform.get_caps to avoid changing other tests that might get the same caps entry.

Reply:
Actuallly, I think it would be better to define a new method in base class that subclass can override to remove unsupported formats.  (see lib/gstreamer/vaapi/vppbase.py for example)

Reply:
> Actuallly, I think it would be better to define a new method in base class that subclass can override to remove unsupported formats. (see lib/gstreamer/vaapi/vppbase.py for example)

Already updated, pls review the patch thanks!

Reply:
To be clear, I mean that the format is not a platform caps limitation, so we should not modify it.  Rather, it is a middleware limitation.  Thus, we should override mapformat and map_best_hw_format which is where middleware specific support formats are selected.

Reply:
> To be clear, I mean that the format is not a platform caps limitation, so we should not modify the caps. Rather, it is a middleware limitation. Thus, we should override mapformat and map_best_hw_format which is where middleware specific supported formats are selected.

updated, thanks.